plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -932,10 +952,38 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
 
         try {
             TrinoFileSystem fileSystem = fileSystemFactory.create(session);
-            if (!fileSystem.listFiles(deltaLogDirectory).hasNext()) {
+            boolean transactionLogFileExists = fileSystem.listFiles(deltaLogDirectory).hasNext();
+
+            if (!replaceExistingTable && transactionLogFileExists) {
+                throw new TrinoException(
+                        NOT_SUPPORTED,
+                        "Using CREATE TABLE with an existing table content is disallowed, instead use the system.register_table() procedure.");
+            }
+            else {
+                long commitVersion = 0;
                 TransactionLogWriter transactionLogWriter = transactionLogWriterFactory.newWriterWithoutTransactionIsolation(session, location);
+                List<AddFileEntry> activeFiles = ImmutableList.of();
+                ProtocolEntry protocolEntry;
+                if (replaceExistingTable) {
+                    commitVersion = getMandatoryCurrentVersion(fileSystem, location) + 1;
+                    transactionLogWriter = transactionLogWriterFactory.newWriter(session, location);
+                    activeFiles = transactionLogAccess.getActiveFiles(
+                            getSnapshot(session, tableHandle),
+                            tableHandle.getMetadataEntry(),
+                            tableHandle.getProtocolEntry(),
+                            tableHandle.getEnforcedPartitionConstraint(),
+                            tableHandle.getProjectedColumns(),
+                            session);
+                    protocolEntry = protocolEntryForTable(tableHandle.getProtocolEntry(), containsTimestampType, tableMetadata.getProperties());
+                    statisticsAccess.deleteExtendedStatistics(session, schemaTableName, location);
+                }
+                else {
+                    setRollback(() -> deleteRecursivelyIfExists(fileSystem, deltaLogDirectory));
```

Comment:
I'm assuming that this branch is for the situation where the transaction log exists , but we're not dealing with a `CREATE OR REPLACE TABLE` statement.

This is however handled through the check

```
if (!replaceExistingTable && transactionLogFileExists) {
                throw new TrinoException(
                        NOT_SUPPORTED,
                        "Using CREATE TABLE with an existing table content is disallowed, instead use the system.register_table() procedure.");
}
```